packages/lesswrong/server/resolvers/subscribedUsersFeedResolver.tsx

```diff
@@ -0,0 +1,78 @@
+import { Posts } from "../../lib/collections/posts";
+import { Subscriptions } from "../../lib/collections/subscriptions";
+import { defineFeedResolver } from "../utils/feedUtil";
+import { addGraphQLSchema } from "../vulcan-lib";
+import { Comments } from "../../lib/collections/comments";
+import keyBy from "lodash/keyBy";
+import { PostsRepo } from "../repos";
+import { loadByIds } from "../../lib/loaders";
+import { filterNonnull } from "../../lib/utils/typeGuardUtils";
+import { accessFilterMultiple } from "../../lib/utils/schemaUtils";
+
+addGraphQLSchema(`
+  type SubscribedPostAndComments {
+    _id: String!
+    post: Post!
+    comments: [Comment!]!
+    postIsFromSubscribedUser: Boolean!
+  }
+`);
+
+defineFeedResolver<Date>({
+  name: "SubscribedFeed",
+  cutoffTypeGraphQL: "Date",
+  args: "af: Boolean",
+  resultTypesGraphQL: `
+    postCommented: SubscribedPostAndComments
+  `,
+  resolver: async ({limit=20, cutoff, offset, args, context}: {
+    limit?: number, cutoff?: Date, offset?: number,
+    args: {af: boolean},
+    context: ResolverContext
+  }) => {
+    const currentUser = context.currentUser;
+    if (!currentUser) throw new Error("Must be logged in");
+
+    const postsAndCommentsAll = await new PostsRepo().getPostsAndCommentsFromSubscriptions(currentUser._id, 30);
+    const postsAndComments = postsAndCommentsAll.slice(offset, (offset??0)+limit);
+    const isLastPage = postsAndComments.length < limit;
+    
+    const postIds: string[] = postsAndComments.map(row => row.postId);
+    const commentIds: string[] = postsAndComments.flatMap(row => row.commentIds ?? []);
+    
+    const [posts, comments] = await Promise.all([
+      (async () =>
+        accessFilterMultiple(currentUser, Posts,    await loadByIds(context, "Posts",    postIds),    context)
+      )(),
+      (async () =>
+        accessFilterMultiple(currentUser, Comments, await loadByIds(context, "Comments", commentIds), context)
+      )()
```

Comment:
wat (also, spacing??)

Reply:
oh, I see, the closures are needed for the inlined `await loadByIds`?  In that case I think it'd probably be more readable to do something like... `loadByIds(...).then(posts => accessFilterMultiple(...))` instead.